controllers/cluster_controller.go

```diff
@@ -120,15 +121,33 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 		return ctrl.Result{}, err
 	}
 
-	if !cluster.ObjectMeta.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(ctx, cluster)
+	return ctrl.Result{}, nil
+}
+
+// reconcile handles cluster reconciliation.
+func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster) (err error) {
+	// If object doesn't have a finalizer, add one.
+	if !util.Contains(cluster.Finalizers, clusterv1.ClusterFinalizer) {
+		cluster.Finalizers = append(cluster.ObjectMeta.Finalizers, clusterv1.ClusterFinalizer)
 	}
 
-	return ctrl.Result{}, nil
+	// TODO(vincepri): These can be generalized with an interface and possibly a for loop.
+	errors := []error{}
+	errors = append(errors, r.reconcileInfrastructure(ctx, cluster))
```

Comment:
artifact from refactoring? It doesn't look like errors is ever larger than a single item (the result from `r.reconcileInfrastructure`)

Reply:
Yeah, but we'll add the `reconcileSecrets` soon I guess, so I left it there